pkg/utils/filesystem.go

```diff
@@ -69,10 +67,37 @@ func CreateDirIfNotExist(path string, perm os.FileMode) error {
 	return err
 }
 
+// IsDirectory returns true if the given path is a directory, false otherwise
 func IsDirectory(path string) (bool, error) {
 	fileInfo, err := os.Stat(path)
 	if err != nil {
 		return false, err
 	}
 	return fileInfo.IsDir(), err
 }
+
+// EnsureLink ensures that a link to the source exists at the target
+// Both source and target should be full paths
+func EnsureLink(source, target string) error {
+
+	// Deal with the file already existing
+	if data, err := os.Lstat(target); !os.IsNotExist(err) {
+		if data.Mode()&os.ModeSymlink != 0 {
+			linkPath, err := os.Readlink(target)
+			if err != nil {
+				return err
+			}
+
+			// Already pointing to the right place, nothing to do
+			if source == linkPath {
+				return nil
+			}
```

Comment:
we should probably do something in the else condition here, where the link is pointing to the wrong spot

Reply:
Fixed - added os.Remove after this to remove the existing symlink if it wasn't right.  Also added checks to catch errors on os.Remove